src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_blocking.cpp

```diff
@@ -48,27 +48,129 @@ bool BrgemmBlocking::run(snippets::lowered::LinearIR& linear_ir) {
 
     bool modified = false;
     for (auto expr_it = linear_ir.begin(); expr_it != linear_ir.end(); expr_it++) {
-        const auto& expr = *expr_it;
-        const auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(expr->get_node());
-        if (!brgemm || blocking_loop_exists(expr, brgemm))
+        const auto& brgemm_expr = *expr_it;
+        const auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(brgemm_expr->get_node());
+        if (!brgemm || blocking_loop_exists(brgemm_expr, brgemm))
             continue;
 
-        const auto& input_shape_0 = expr->get_input_port_descriptor(0)->get_shape();
-        const auto& input_layout_0 = expr->get_input_port_descriptor(0)->get_layout();
-        const auto& dim = *(input_layout_0.rbegin() + dim_idx);
-        const auto& m = input_shape_0[dim];
+        const auto& in_0_desc = brgemm_expr->get_input_port_descriptor(0);
+        const auto& in_1_desc = brgemm_expr->get_input_port_descriptor(1);
+        const auto& out_desc = brgemm_expr->get_output_port_descriptor(0);
+
+        const auto& in_0_planar_dims = ov::snippets::utils::get_planar_vdims(in_0_desc->get_shape(), in_0_desc->get_layout());
+        const auto& in_1_planar_dims = ov::snippets::utils::get_planar_vdims(in_1_desc->get_shape(), in_1_desc->get_layout());
+        const auto& out_preordered_dims = ov::snippets::utils::get_preordered_vdims(out_desc->get_shape(), out_desc->get_layout());
+
+        auto in_0_subtensor = in_0_desc->get_subtensor();
+        auto in_1_subtensor = in_1_desc->get_subtensor();
+        auto out_subtensor = out_desc->get_subtensor();
+
+        auto apply_m_blocking = [&]() {
+            const auto& m = *(out_preordered_dims.rbegin() + 1);
+            const auto block_size_m = brgemm->get_m_block_size();
+            if (block_size_m >= m) {
+                *(in_0_subtensor.rbegin() + 1) = m;
+                *(out_subtensor.rbegin() + 1) = m;
+            } else {
+                *(in_0_subtensor.rbegin() + 1) = block_size_m;
+                *(out_subtensor.rbegin() + 1) = block_size_m;
+
+                std::vector<LoopPort> entries{LoopPort(brgemm_expr->get_input_port(0), true),
+                                              LoopPort(brgemm_expr->get_input_port(1), false)};
+                if (brgemm->is_with_scratchpad())
+                    entries.emplace_back(brgemm_expr->get_input_port(2), false);
```

Comment:
I'd like to remind to you the AMX case with new memory buffer when this buffer should be inside all blocking loops since the memory is needed only in brgemm kernel evaluation (it's not shared between loop iterations).

Actually I implemented this [case](https://github.com/openvinotoolkit/openvino/blob/1ac5e1f6bb4d81a291985d09d38d8b1c4f00228a/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_blocking.cpp#L83) in the my [PR](https://github.com/openvinotoolkit/openvino/pull/19644). Could you please move the new memory buffer to loops in this PR as well?

Reply:
Thanks for the suggestion, I reused your implementation for all blocking loops